proposals/pubsub.md

```diff
@@ -0,0 +1,182 @@
+# PubSub enabled by default
+
+Authors:
+- @gozala
+
+Initial PR: https://github.com/protocol/web3-dev-team/pull/53
+
+<!--
+This template is for a proposal/brief/pitch for a significant project to be undertaken by a Web3 Dev project team.
+The goal of project proposals is to help us decide which work to take on, which things are more valuable than other things.
+-->
+<!--
+A proposal should contain enough detail for others to understand how this project contributes to our team’s mission of product-market fit
+for our unified stack of protocols, what is included in scope of the project, where to get started if a project team were to take this on,
+and any other information relevant for prioritizing this project against others.
+It does not need to describe the work in much detail. Most technical design and planning would take place after a proposal is adopted.
+Good project scope aims for ~3-5 engineers for 1-3 months (though feel free to suggest larger-scoped projects anyway). 
+Projects do not include regular day-to-day maintenance and improvement work, e.g. on testing, tooling, validation, code clarity, refactors for future capability, etc.
+-->
+<!--
+For ease of discussion in PRs, consider breaking lines after every sentence or long phrase.
+-->
+
+
+
+## Purpose &amp; impact 
+#### Background &amp; intent
+_Describe the desired state of the world after this project? Why does that matter?_
+<!--
+Outline the status quo, including any relevant context on the problem you’re seeing that this project should solve.
+Wherever possible, include pains or problems that you’ve seen users experience to help motivate why solving this problem works towards top-line objectives.
+-->
+
+As of this writing [pubsub][IPFS pubsub] is considered experimental and is disabled in go-ipfs.
```

Comment:
It would be great if we could have some background in here about _why_ it's still experimental, is it to do with the uncertainty hinted at in the plan of attack, that overhead could create problems if activated en mass?

Reply:
https://github.com/ipfs/go-ipfs/issues/6621

Reply:
```suggestion
As of this writing [pubsub][IPFS pubsub] is considered experimental and is disabled in go-ipfs (due to increased resource usage on nodes that do not make use of it).
```

@rvagg @aschmahmann does this clarify and is accurate ?

Reply:
will let Adin weigh in further if he thinks needed, but it might be worth pointing to that issue in your proposal, specifically @Stebalien points to two more potential blockers that may (or may not) need resolving to get this to happen, which will impact scope of this work https://github.com/ipfs/go-ipfs/issues/6621#issuecomment-642846010

Reply:
It may also be at a point where it's "mostly not a problem" and we can just enable it. We should try pulling some profiles from lotus to see how much memory/CPU it uses as that'll give us the best estimate of the cost.

Ideally, I'd _like_ to have an "enable on demand" kind of system, but that will require a bit of work. Specifically, pubsub would need to respond to protocol changes from libp2p.

Reply:
> Ideally, I'd like to have an "enable on demand" kind of system, but that will require a bit of work. Specifically, pubsub would need to respond to protocol changes from libp2p.

Thinking about it got me wondering if this going to make network sparse enough to make pubsub unreliable, especially web nodes, which may not even have a connection(s) that actively participate in pubsub.

Reply:
Nodes only participate in pubsub channels that they're subscribed to anyways. Basically, a pubsub "network" is composed of all nodes participating in a particular pubsub _topic_.